lib/vector-core/src/event/vrl_target.rs

```diff
@@ -245,40 +253,52 @@ impl vrl_core::Target for VrlTarget {
                 }
 
                 Ok(None)
+                    */
             }
         }
     }
 }
 
-impl From<Event> for VrlTarget {
-    fn from(event: Event) -> Self {
-        VrlTarget::new(event)
-    }
-}
-
 // Turn a `Value` back into `LogEvents`:
 // * In the common case, where `.` is a map, just create an event using it as the event fields.
 // * If `.` is an array, map over all of the values to create log events:
 //   * If an element is an object, create an event using that as fields.
 //   * If an element is anything else, assign to the `message` key.
 // * If `.` is anything else, assign to the `message` key.
-fn value_into_log_events(value: Value, metadata: EventMetadata) -> impl Iterator<Item = Event> {
-    match value {
-        Value::Map(object) => Box::new(std::iter::once(Event::from(LogEvent::from_parts(
-            object, metadata,
-        )))) as Box<dyn Iterator<Item = Event>>,
-        Value::Array(values) => Box::new(values.into_iter().map(move |v| match v {
-            Value::Map(object) => Event::from(LogEvent::from_parts(object, metadata.clone())),
-            v => {
-                let mut log = LogEvent::new_with_metadata(metadata.clone());
-                log.insert(log_schema().message_key(), v);
-                Event::from(log)
+fn value_into_log_events(
+    value: SharedValue,
+    metadata: EventMetadata,
+) -> impl Iterator<Item = Event> {
+    match &*value.borrow() {
+        vrl_core::Value::Object(object) => {
+            Box::new(std::iter::once(Event::from(LogEvent::from_parts(
+                object
+                    .iter()
+                    .map(|(k, v)| (k.clone(), v.borrow().clone().into()))
+                    .collect(),
```

Comment:
There is a cost here since we now need to convert back from a `vrl::Value` into a Vector `Value`. 

This is one downside of the approach.